cli/src/stake.rs

```diff
@@ -708,20 +746,21 @@ pub fn process_deactivate_stake_account(
     let (nonce_authority, nonce_authority_pubkey) = nonce_authority
         .map(|a| (a.keypair(), a.pubkey()))
         .unwrap_or((&config.keypair, config.keypair.pubkey()));
+    let fee_payer = fee_payer.map(|f| f.keypair()).unwrap_or(&config.keypair);
     let mut tx = if let Some(nonce_account) = &nonce_account {
         Transaction::new_signed_with_nonce(
             ixs,
-            Some(&config.keypair.pubkey()),
-            &[&config.keypair, nonce_authority, stake_authority],
+            Some(&fee_payer.pubkey()),
+            &[fee_payer, nonce_authority, stake_authority],
             nonce_account,
             &nonce_authority.pubkey(),
             recent_blockhash,
         )
     } else {
         Transaction::new_signed_with_payer(
             ixs,
-            Some(&config.keypair.pubkey()),
-            &[&config.keypair, stake_authority],
+            Some(&fee_payer.pubkey()),
+            &[fee_payer, stake_authority],
```

Comment:
with this change, what is the purpose of `config.keypair`?

Reply:
or: why do these commands take fee_payer, when you could accomplish the same with `--keypair` (already a global, top-level argument)?

Reply:
> with this change, what is the purpose of `config.keypair`?

It is still the fallback for all signing authorities

> or: why do these commands take fee_payer, when you could accomplish the same with `--keypair` (already a global, top-level argument)?

Two-fold:
1. I'm not ready to impose the `SigningAuthority` concept globally. It'll require a more careful review of all of the usages which I don't think I can prioritize right now
2. Improved conciseness as to signer roles

Reply:
@rob-solana to elaborate [2], there are several places where the "operative account" keypair and fee-payer keypair are assumed to be the same, but this can be undesirable.   For instance, sweeping an account balance would be simpler if the fee-payer were able to be specified independently from the `from` account. This saves the user the fee calculation and subtraction from balance, or worse, us maintaining another subcommand.

Reply:
@rob-solana any action I can take here?

Reply:
I think the implicit --keypair argument is probably the biggest impediment to sanity for these operations.  Were you to remove that logic and have each argument spelled out (with defaults for common cases), we'd be better off.

Reply:
So you're looking for something more along these lines?
```
solana <SUBCOMMAND> <OPERATIVE_ACCOUNT> [FEE_ACCOUNT] [SUBCOMMAND_ARG...]
```
I'd tend to agree (killing any implicit behavior, really) and I'm totally up for that, but I *really* need to get on to snapshot verification, so was hoping for the minimum to get us functional offline signing here.  Is this something we could plan more thoroughly and build out once I'm through with snapshot verification? We're going to need further changes to allow for mixed absent/present signers anyway and I suspect @CriesofCarrots will need some tweaks to support hardware wallets as well.

Reply:
Maybe tag @CriesofCarrots into the ring?  Totally agree, @t-nelson please report to snapshot verification.  ⏩

Reply:
I'm definitely going to want a configured (implicit) key for remote wallet integration, and I generally think it makes sense to have for basic actions, with optional overrides.

We will need to make some fundamental changes to way we're passing keypairs around in order to integrate remote wallet signing, but I'm able to implement remote wallet key reads in a fairly lightweight fashion now. @t-nelson , I can probably take over those changes so that you can focus on snapshots, but I definitely would like a chat with you about the refactoring you've been considering.

Reply:
@CriesofCarrots that'd be great! I've got #8052 and #8108 backed up behind this one. Dan needs these for his offline/nonce testing. They should JustGo(TM) +/- review if we can put this in. Save us the single-use knowledge transfer to get rebases in order.

I'd be happy to chat about the changes I have in mind. We can do it Friday in person or schedule a call if you think you'll want to get started sooner!